speech/transcribeStreaming.v2.js

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// sample-metadata:
+//   title: Transcribe a local file using streaming recognition
+//   description: Streaming transcription of a local file using Speech-to-Text v2
+
+async function main(recognizerName, filename = 'resources/brooklyn.flac') {
+  //[START speech_transcribe_streaming_v2]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const recognizerName = 'Resource name, e.g. projects/[projectId]/locations/global/recognizers/[recognizerId]'
+  // const filename = 'Local path to audio file, e.g. /path/to/audio.raw';
+
+  const fs = require('fs');
+
+  // Imports the Google Cloud client library
+  const speech = require('@google-cloud/speech').v2;
+
+  async function streamingRecognize() {
+    // Creates a client
+    const client = new speech.SpeechClient();
+
+    const recognitionConfig = {
+      // autoDecodingConfig removes the need to specify audio encoding.
+      // This field only needs to be present in the recognitionConfig
+      autoDecodingConfig: {},
+    };
+    const streamingConfig = {
+      config: recognitionConfig,
+    };
+
+    const configRequest = {
+      recognizer: recognizerName,
+      streamingConfig: streamingConfig,
+    };
+
+    // Read a file as bytes, base64 encoded
+    const readStream = fs.createReadStream(filename, {
+      highWaterMark: 4096,
+      encoding: 'base64',
+    });
+
+    const chunks = [];
+    readStream
+      .on('data', chunk => {
+        const request = {
+          audio: chunk.toString(),
+        };
+        chunks.push(request);
+      })
+      .on('close', () => {
+        // Config-only request should be first in stream of requests
+        stream.write(configRequest);
+        for (const chunk of chunks) {
+          stream.write(chunk);
+        }
+        stream.end();
+      });
+
+    const stream = client
```

Comment:
We should use `Promise.{resolve, reject}()` and `await` here, instead of returning `stream`.

It might look something like this:
```
await new Promise((resolve, reject) => {
  client
      ._streamingRecognize()
      .on('data', response => {
          const {results} = response;
          const {transcript} = results[0].alternatives[0];
          return Promise.resolve(transcript);
      })
      .on('error', err => {
          /* omit console.error() here - rejecting a Promise here is equivalent to throwing an exception */
          return Promise.reject(err);
      })

console.log(transcript);
```

Reply:
Done. I like this way a lot better!